### DISCUSSION TITLE
submitting form refreshes page instead of executing onSubmit function.

### DISCUSSION THREAD

**Discussion by adammo94** [react-hook-form/react-hook-form#7917] [Q&A] [answered]
I have a form on one page that behaves strangely, I literally copied it from other component in my app where it works fine.
On this specific page when I click button (which should submit the form) instead of executing onSubmit function page tries to submit the form with get request which leads to adding `?` in URL. I literally commented out all my inputs, but it still refreshes.

```
const onSubmit = async (data: any) => {
  console.log(data);
};
```
```
const {
    register,
    watch,
    handleSubmit,
    control,
    reset,
  } = useForm<FieldValues>();
```

```
<ActiveDecisionWrapper
  noValidate
  onSubmit={handleSubmit(onSubmit)}
>
```
ActiveDecisionWrapper is a form.
And inside it I have:
```
<Button variant="isPrimary">
  Save changes
</Button>
```

**Comment by adammo94** [comment, marked as answer]
Classical me, found out I had two forms one inside another.

**Reply by GourishettySairam** [reply]
faced the same issue, One of my form is wrapped in styled components and i overlooked it. Thanks for posting this, it really helped me.

**Reply by zhondori** [reply]
Thank you for the answer, but how can i fix this issue? I also have two form one inside another but what is the solution?

**Reply by marko-hologram** [reply]
> Thank you for the answer, but how can i fix this issue? I also have two form one inside another but what is the solution?

You must stop nesting forms. HTML spec says that `<form>` elements cannot be nested.

```html
<form>
   First form

  <form>Nested form</form>
</form>
```

This above is **invalid** HTML. 

Solution is to structure your code not to have nested forms. Hard to say what's the appropriate solution without understanding what you are trying to achieve and why would you require nested forms.

**Reply by krishna7889** [reply]
thank you so much

**Reply by Boomin7x** [reply]
i did same thing too, thanks for this

**Comment by shaquan1228** [comment]
Yall are amazing. This was my problem and I spent a very long time looking in every other direction. <img src="https://media0.giphy.com/media/l0HlFZ3c4NENSLQRi/giphy.gif"/>

**Comment by imtheaman** [comment]
Yeah, in my case it was happening cuz of webpack's hot reloading, then i created a build and ran that, things were working fine in release build

** In my case things were working correctly on local desktop environments but not on the local mobiles connected via ip on the same network **

**Comment by oldtune** [comment]
For me it was `<form onSubmit={() => { handleSubmit(props.save); }}>` that caused the issue. Changed it to 

`<form onSubmit={handleSubmit(props.save)}` 

fixed the issue
